src/pages/PullRequestPage/subroute/FilesChangedTab/FilesChanged/Table/FilesChangedTable.tsx

```diff
@@ -0,0 +1,355 @@
+import {
+  createColumnHelper,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import cs from 'classnames'
+import { isNumber } from 'lodash'
+import isEmpty from 'lodash/isEmpty'
+import { Fragment, lazy, Suspense, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import {
+  ImpactedFile,
+  OrderingDirection,
+  OrderingParameter,
+  usePull,
+} from 'services/pull/usePull'
+import A from 'ui/A'
+import Icon from 'ui/Icon'
+import Spinner from 'ui/Spinner'
+import 'ui/FileList/FileList.css'
+import TotalsNumber from 'ui/TotalsNumber'
+
+const PullFileDiff = lazy(() => import('../../shared/FileDiff'))
+
+const columnHelper = createColumnHelper<ImpactedFile>()
+
+const isNumericValue = (value: string) =>
+  value === 'missedLines' ||
+  value === 'patchPercentage' ||
+  value === 'head' ||
+  value === 'change'
+
+export function getFilter(sorting: Array<{ id: string; desc: boolean }>) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state?.desc
+      ? OrderingDirection.desc
+      : OrderingDirection.asc
+
+    let parameter = undefined
+    if (state.id === 'name') {
+      parameter = OrderingParameter.FILE_NAME
+    }
+
+    if (state.id === 'missedLines') {
+      parameter = OrderingParameter.MISSES_COUNT
+    }
+
+    if (state.id === 'patchPercentage') {
+      parameter = OrderingParameter.PATCH_COVERAGE
+    }
+
+    if (state.id === 'head') {
+      parameter = OrderingParameter.HEAD_COVERAGE
+    }
+
+    if (state.id === 'change') {
+      parameter = OrderingParameter.CHANGE_COVERAGE
+    }
+
+    return { direction, parameter }
+  }
+
+  return undefined
+}
+
+function getColumns({ pullId }: { pullId: string }) {
+  return [
+    columnHelper.accessor('headName', {
+      id: 'name',
+      header: 'Name',
+      cell: ({ getValue, row }) => {
+        const headName = getValue()
+
+        return (
+          <div className="flex flex-row break-all">
+            <span
+              data-action="clickable"
+              data-testid="file-diff-expand"
+              className={cs(
+                'inline-flex items-center gap-1 font-sans hover:underline focus:ring-2',
+                {
+                  'text-ds-blue': row.getIsExpanded(),
+                }
+              )}
+              {...{
+                onClick: row.getToggleExpandedHandler(),
+              }}
```

Comment:
`l`: is this object spread required? or are you able to switch it to just a plain old `onClick` handler:

```tsx
<div
  onClick={() => {
    row.getToggleExpandedHandler()
  }}
>
  // ...
</div>
```

Reply:
it doesn't work unfortunately

Reply:
Sorry I messed up my example, should actually be:

```ts
onClick={header.column.getToggleSortingHandler()}
```